File: wallet/src/main/java/io/samos/wallet/activities/Transaction/old/TransactionAdapter.java
package io.samos.wallet.activities.Transaction.old;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.text.DecimalFormat;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import io.samos.wallet.R;
import io.samos.wallet.beans.TransactionInfo;
import io.samos.wallet.common.Constant;
import io.samos.wallet.common.OnItemClickListener;
import io.samos.wallet.utils.SharePrefrencesUtil;

/**
 * Created by kimi on 2018/1/29.</br>
 */

public class TransactionAdapter extends RecyclerView.Adapter<TransactionAdapter.MainTransactionViewHolder> {

    List<TransactionInfo> transactionInfos;
    OnItemClickListener<TransactionInfo> onItemClickListener;

    SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd") ;

    //汇率
    double exchangeCoin = SharePrefrencesUtil.getInstance().getBoolean(Constant.IS_LANGUAGE_ZH)
            ? Double.parseDouble(SharePrefrencesUtil.getInstance().getString(Constant.CNYEXCHAGECOIN))
            : Double.parseDouble(SharePrefrencesUtil.getInstance().getString(Constant.USDEXCHAGECOIN));
    DecimalFormat decimalFormat = new DecimalFormat(".000");

    public TransactionAdapter(List<TransactionInfo> mainTransactionBeanList){
        this.transactionInfos = mainTransactionBeanList == null ? new ArrayList<TransactionInfo>() : mainTransactionBeanList;
    }

    public void setOnItemClickListener( OnItemClickListener<TransactionInfo> onItemClickListener){
        this.onItemClickListener = onItemClickListener;
    }

    public List<TransactionInfo> getDatas(){
        return transactionInfos;
    }

    private View.OnClickListener onClickListener = new View.OnClickListener(){

        @Override
        public void onClick(View v) {
            if(onItemClickListener == null) {
                return;
            }
            int position = (int) v.getTag();
            onItemClickListener.onClick(v.getId(),position,transactionInfos.get(position));
        }
    };

    @Override
    public MainTransactionViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new MainTransactionViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_wallet_transaction,parent,false));
    }

    @Override
    public void onBindViewHolder(MainTransactionViewHolder holder, int position) {
        TransactionInfo transactionInfo = transactionInfos.get(position);
        String time = transactionInfo.getTime();
        Date date = new Date(Long.parseLong(time));
        holder.date.setText(simpleDateFormat.format(date));
        holder.address.setText(transactionInfo.getToWallet());
        holder.transaction.setText("-"+transactionInfo.getAmount());
        holder.flag.setText("  Send "+transactionInfo.getCoinType());
        holder.itemView.setTag(position);
        holder.address.setTag(position);
        holder.address.setOnClickListener(onClickListener);
        holder.itemView.setOnClickListener(onClickListener);
        double value = Double.valueOf(transactionInfo.getAmount()) * exchangeCoin;
        if (SharePrefrencesUtil.getInstance().getBoolean(Constant.IS_LANGUAGE_ZH)) {
            holder.balance.setText("- ￥" + decimalFormat.format(value));
        } else
            holder.balance.setText("- $" + decimalFormat.format(value));
    }

    @Override
    public int getItemCount() {
        return transactionInfos.size();
    }


    /**
     * viewholder
     */
    public class MainTransactionViewHolder extends RecyclerView.ViewHolder{

        public ImageView img;
        public TextView flag,date,address,transaction,balance;

        public MainTransactionViewHolder(View itemView) {
            super(itemView);

            date = itemView.findViewById(R.id.tv_date);
            address = itemView.findViewById(R.id.tv_address);
            balance = itemView.findViewById(R.id.tv_balance);
        }
    }
}

File: wallet/src/main/java/io/samos/wallet/activities/Transaction/TransactionAdapter.kt
package io.samos.wallet.activities.Transaction

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import io.samos.wallet.R
import io.samos.wallet.beans.TransInfo
import io.samos.wallet.beans.TransactionInfo
import io.samos.wallet.common.Constant
import kotlinx.android.synthetic.main.dialog_rollout_payment_detail.view.*
import kotlinx.android.synthetic.main.item_wallet_transaction.view.*
import java.text.SimpleDateFormat
import java.util.*

class TransactionAdapter(var context: Context, var transaction: ArrayList<TransInfo>?) : RecyclerView
.Adapter<RecyclerView.ViewHolder>() {
    /**
     * 回调接口
     */
    var TransactionListener: OnItemClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder? {
        return MainWalletViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_wallet_transaction, parent, false))
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        holder.itemView.tag = position
        var trans: TransInfo = transaction?.get(position)!!

        holder.itemView.setOnClickListener {
            TransactionListener?.OnClick(trans)
        }

        holder.itemView.tv_amount.text = trans.delta
        holder.itemView.tv_state.text = trans.state



        val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
        var str = trans.time+"000"


        val dt = dateFormat.format(str.toLong())
        holder.itemView.tv_date.text = dt
        holder.itemView.tv_address.text = trans.txid
    }

    override fun getItemCount(): Int {
        return transaction?.size ?: 0
    }


    /**
     * 钱包对应的holder
     */
    class MainWalletViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tv_address: TextView
        var tv_state: TextView
        var tv_amount: TextView
        var tv_date: TextView

        init {
            tv_address = itemView.findViewById(R.id.tv_address)
            tv_state = itemView.findViewById(R.id.tv_state)
            tv_amount = itemView.findViewById(R.id.tv_amount)
            tv_date = itemView.findViewById(R.id.tv_date)

        }

    }

    interface OnItemClickListener {
        fun OnClick(trans: TransInfo?)
    }
}